### PR TITLE
feat(infinite-hits): return banner info with connectInfiniteHits

### PR DESCRIPTION
**Pull request by taylorcjohnson**
**Summary**

Following on from the recent addition of the rules controlled banner to `Hits`, we're adding the same to `InfiniteHits`. This first PR returns banner information with `connectInfiniteHits`.

https://algolia.atlassian.net/browse/EMERCH-1509

**Result**

Updated and passing test (no UI changes)

### CODE REVIEW

File: packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ export type InfiniteHitsConnectorParams<
   cache?: InfiniteHitsCache<THit>;
 };
 
+type Banner = NonNullable<
+  NonNullable<
+    Required<SearchResults<Hit>['renderingContent']>
+  >['widgets']['banners']
+>[number];
+
```
Comment (taylorcjohnson) [inline packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts:93]: Should this definition (and the same definition in `connectHits.ts`) be moved to `packages/instantsearch.js/src/types.results.ts`?

Comment (aymeric-giraudet) [reply packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts:93]: Yes 👍
Looks good otherwise

**Review by aymeric-giraudet — APPROVED** [review]

**Review by Haroenv — APPROVED** [review]
code is correct

File: packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts [hunk 2]
```diff
@@ -85,6 +85,12 @@ export type InfiniteHitsConnectorParams<
   cache?: InfiniteHitsCache<THit>;
 };
 
+type Banner = NonNullable<
+  NonNullable<
+    Required<SearchResults<Hit>['renderingContent']>
+  >['widgets']['banners']
+>[number];
```
[the remark this answers is not in the document]
Comment (taylorcjohnson) [reply packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts:92]: Sounds good. To be clear, you are suggesting we import it from: `/instantsearch/packages/algoliasearch-helper/types/algoliasearch.d.ts`

[the remark this answers is not in the document]
Comment (Haroenv) [reply packages/instantsearch.js/src/connectors/infinite-hits/connectInfiniteHits.ts:92]: I'm more thinking the index.d.ts there, what do you think?

**Review by Haroenv — APPROVED** [review]

**Pull request merged**